services/device-registry-mongodb/src/test/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedTenantServiceTest.java

```diff
@@ -84,25 +81,7 @@ public void setup(final TestInfo testInfo) {
      */
     @AfterEach
     public void cleanCollection(final VertxTestContext testContext) {
-        mongoClient.removeDocuments(
-                config.getCollectionName(),
-                new JsonObject(),
-                testContext.completing());
-    }
-
-    /**
-     * Cleans up fixture.
-     *
-     * @param testContext The test context to use for running asynchronous tests.
-     */
-    @AfterAll
-    public void finishTest(final VertxTestContext testContext) {
-
-        mongoClient.close();
```

Comment:
Given that we are using a shared connection pool, I do not really see what could be leaked here. Once the vert.x instance is being closed (i.e. the service process is terminated), the connections in the pool will be closed implicitly, right?
I would rather not let the DAOs implement Lifecycle because I do not want to tie the DAOs to the Verticle lifecycle. Instead, I will add a *closeMongoClient* method to *MongoDbBasedDao* and annotate it with `@PreDestroy`. That method can then be used by the tests and will be invoked automatically by the Spring bean container when shutting down the application.